src/Atlas2DSlice.ts

```diff
@@ -60,16 +60,30 @@ export default class Atlas2DSlice implements VolumeRenderImpl {
 
     this.geometryTransformNode.add(this.boxHelper, this.geometryMesh);
 
-    this.setUniform("ATLAS_X", volume.imageInfo.cols);
-    this.setUniform("ATLAS_Y", volume.imageInfo.rows);
-    this.setUniform("textureRes", new Vector2(volume.imageInfo.atlas_width, volume.imageInfo.atlas_height));
-    this.setUniform("SLICES", volume.z);
-    this.setUniform("Z_SLICE", Math.floor(volume.z / 2));
-
-    this.channelData = new FusedChannelData(volume.imageInfo.atlas_width, volume.imageInfo.atlas_height);
+    this.setUniform("Z_SLICE", Math.floor((volume.imageInfo.vol_size_z || volume.z) / 2));
+    this.initChannelData();
     this.updateSettings(settings, SettingsFlags.ALL);
   }
 
+  private initChannelData() {
+    /* eslint-disable-next-line @typescript-eslint/naming-convention */
+    const { cols, rows, tile_width, tile_height } = this.volume.imageInfo;
+    const atlasWidth = tile_width * cols;
+    const atlasHeight = tile_height * rows;
+
+    this.setUniform("ATLAS_X", cols);
+    this.setUniform("ATLAS_Y", rows);
+    this.setUniform("textureRes", new Vector2(atlasWidth, atlasHeight));
+    this.setUniform("SLICES", this.volume.imageInfo.vol_size_z || this.volume.z);
+    this.setUniform("SUBSET_SCALE", this.volume.contentSize);
+    this.setUniform("SUBSET_OFFSET", this.volume.contentOffset);
+
+    if (!this.channelData || this.channelData.width !== atlasWidth || this.channelData.height !== atlasHeight) {
+      this.channelData?.cleanup();
+      this.channelData = new FusedChannelData(atlasWidth, atlasHeight);
+    }
+  }
```

Comment:
This function and an equivalent one on `RayMarchedAtlasVolume` currently run twice when the constructor is called. I should think hard about how to make this no longer the case.